homeassistant/components/reolink/binary_sensor.py

```diff
@@ -198,3 +359,42 @@ async def async_added_to_hass(self) -> None:
     async def _async_handle_event(self, event: str) -> None:
         """Handle incoming event for motion detection."""
         self.async_write_ha_state()
+
+
+class ReolinkSmartAIBinarySensorEntity(
+    ReolinkChannelCoordinatorEntity, BinarySensorEntity
+):
+    """Binary-sensor class for Reolink IP camera smart AI sensors."""
+
+    entity_description: ReolinkSmartAIBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
+        location: int,
+        entity_description: ReolinkSmartAIBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize Reolink binary sensor."""
+        self.entity_description = entity_description
+        super().__init__(reolink_data, channel)
+        unique_index = self._host.api.baichuan.smart_ai_index(
```

Comment:
Are the indexes stable? For example, if the user has 3 sensors (indexes 0,1,2) and deletes the one with index 1, does the other two keep their indexes (0 and 2)? Or does 2 become 1 now?

Reply:
Yes they are stable.
The "location" (also a index) changes, but the "index" is always increasing.
So if you have 3 zones: loc 1 idx 1, loc 2 idx 2, loc 3  idx 3.
Then remove loc 2 you get: loc 1 idx 1, loc 3  idx 3
If you then add a new zone you get: loc 1 idx 1, loc 3  idx 3, loc 2  idx 4

etc. etc.
Eventually after removing and adding a bit you get large numbers for the index like 12 while the location always stays between 1-3.